### PR TITLE
wootility: 4.6.21 -> 4.7.2; wooting-udev-rules: update to latest recommended rules

### PR DESCRIPTION
**Pull request by returntoreality**
Updated both the udev rules and the wootility. The simplified udev rules from the vendor now use uaccess tag which uses logind to give access to logged in users instead of giving access to the input group. Updated the module description to remove mention of the input group. Wootility was updated to the latest stable release.
Added myself as a maintainer to both of them.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by LordGrimmauld** [comment]
Package compiles and runs just fine. Also updates firmware of my W2. I am not exactly sure with the udev rules though, how is one supposed to check for regressions there, caused by imprecise matching?

**Comment by returntoreality** [comment]
Thanks for testing. The udev rules are more general than before, all ATTRS{idVendor}=="03eb" matches are still there, and the matches for ATTRS{idVendor}=="31e3" are no longer restricted to any product so matches all products for that vendor (which is Wooting).
The source is their official help page, which was also the source for the rules before:
https://help.wooting.io/article/147-configuring-device-access-for-wootility-under-linux-udev-rules

**Comment by LordGrimmauld** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 357836`

---
### `x86_64-linux`
<details>
  <summary>:fast_forward: 2 packages blacklisted:</summary>
  <ul>
    <li>nixos-install-tools</li>
    <li>tests.nixos-functions.nixos-test</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>wootility</li>
    <li>wooting-udev-rules</li>
  </ul>
</details>

**Review by LordGrimmauld — APPROVED** [review]

File: pkgs/by-name/wo/wooting-udev-rules/package.nix [hunk 1]
```diff
@@ -2,9 +2,9 @@
 
 stdenv.mkDerivation rec {
   pname = "wooting-udev-rules";
-  version = "unstable-2023-03-31";
+  version = "unstable-2024-11-20";
```
Comment (LordGrimmauld) [inline pkgs/by-name/wo/wooting-udev-rules/package.nix:5]: Nit: Package versions are now supposed to be prefixed with `0-unstable...` so when an eventual release (1.x) comes it'll be picked up as newer.

Comment (returntoreality) [reply pkgs/by-name/wo/wooting-udev-rules/package.nix:5]: Ah good to know

Comment (LordGrimmauld) [reply pkgs/by-name/wo/wooting-udev-rules/package.nix:5]: it should be fine here, as there is no dedicated package containing these udev rules specifically and not looking like there ever will be one. And its previously been like this too. So not a blocker in my opinion, but worth mentioning.

**Review by sodiboo — APPROVED** [review]

**Review by LordGrimmauld — APPROVED** [review]

File: nixos/modules/hardware/wooting.nix [hunk 2]
```diff
@@ -1,7 +1,6 @@
 { config, lib, pkgs, ... }:
 {
-  options.hardware.wooting.enable = lib.mkEnableOption ''support for Wooting keyboards.
-    Note that users must be in the "input" group for udev rules to apply'';
+  options.hardware.wooting.enable = lib.mkEnableOption "Support for Wooting keyboards.";
```
Comment (donovanglover) [inline nixos/modules/hardware/wooting.nix:3]:
```suggestion
  options.hardware.wooting.enable = lib.mkEnableOption "support for Wooting keyboards";
```
`mkEnableOption` makes the sentence automatically

https://github.com/NixOS/nixpkgs/blob/1b0e57933c5cc61ebae72e4d977809c8979f4a7e/lib/options.nix#L98-L105

File: pkgs/by-name/wo/wooting-udev-rules/package.nix (hunk 1 above)
Comment (returntoreality) [reply pkgs/by-name/wo/wooting-udev-rules/package.nix]: Fixed it

File: nixos/modules/hardware/wooting.nix (hunk 2 above)
Comment (returntoreality) [reply nixos/modules/hardware/wooting.nix]: Goodf point, I fixed it

**Review by donovanglover — APPROVED** [review]

**Review by LordGrimmauld — APPROVED** [review]

**Pull request merged**